src/map/services/index.ts

```diff
@@ -2,7 +2,74 @@ import environment from '../../environment'
 
 const mapBaseLayers = require('../../../public/static/map/map-base-layers.config.json')
 
-async function fetchMap(resolver, query) {
+// TODO: Generate these types from the GraphQL schema.
+export interface MapLayer {
+  __typename: 'MapLayer'
+  id: string
+  title: string
+  type: MapLayerType
+  noDetail: boolean
+  minZoom: number
+  maxZoom: number
+  layers?: string[]
+  url?: string
+  params?: string
+  detailUrl: string
+  detailParams?: DetailParams
+  detailIsShape?: boolean
+  iconUrl?: string
+  imageRule?: string
+  notSelectable: boolean
+  external?: boolean
+  bounds: [number[]]
+  authScope?: string
+  category?: string
+  legendItems?: MapLayerLegendItem[]
+  meta: Meta
+  href: string
+}
+
+export interface LegendItem {
+  __typename: 'LegendItem'
+  title: string
+  iconUrl?: string
+  imageRule?: string
+  notSelectable: boolean
+}
+
+export type MapLayerLegendItem = MapLayer | LegendItem
+
+export type MapLayerType = 'wms' | 'tms'
+
+export interface DetailParams {
+  item: string
+  datasets: string
+}
+
+export interface Meta {
+  description?: string
+  themes: Theme[]
+  datasetIds?: number[]
+  thumbnail?: string
+  date?: string
+}
+
+export interface Theme {
+  id: string
+  title: string
+}
+
+export interface MapCollection {
```

Comment:
Could this not be extended like so?:
```suggestion
export interface MapCollection extends Theme {
```